src/test/java/org/isf/medicalsinventory/Tests.java

```diff
@@ -572,4 +575,63 @@ void testDeleteInventory() throws Exception {
 		assertThat(deletedInventory).isNotNull();
 		assertThat(deletedInventory.getStatus()).isEqualTo(InventoryStatus.canceled.toString());
 	}
+	
+	@Test
+	void testValidateMedicalInventory() throws Exception {
+		Ward ward = testWard.setup(false);
+		wardIoOperationRepository.saveAndFlush(ward);
+		MedicalInventory inventory = testMedicalInventory.setup(ward, false);
+		MedicalInventory savedInventory = medicalInventoryIoOperation.newMedicalInventory(inventory);
+		MedicalType medicalType = testMedicalType.setup(false);
+		medicalTypeIoOperationRepository.saveAndFlush(medicalType);
+		Medical medical = testMedical.setup(medicalType, false);
+		medicalsIoOperationRepository.saveAndFlush(medical);
+		Lot lot = testLot.setup(medical, false);
+		lotIoOperationRepository.saveAndFlush(lot);
+		MedicalInventoryRow medicalInventoryRow = testMedicalInventoryRow.setup(savedInventory, medical, lot, false);
+		int inventoryRowId = medicalInventoryRow.getId();
+		MedicalInventoryRow newMedicalInventoryRow = medicalInventoryRowManager.newMedicalInventoryRow(medicalInventoryRow);
+		assertThat(newMedicalInventoryRow).isNotNull();
+		List<MedicalInventoryRow> medicalInventoryRows = medicalInventoryRowManager.getMedicalInventoryRowByInventoryId(inventoryRowId);
+		assertThat(medicalInventoryRows).isNotEmpty();
+		assertThat(medicalInventoryRows).hasSize(1);
+		medicalInventoryManager.validateMedicalInventoryRow(savedInventory, medicalInventoryRows);
+		int inventoryId = inventory.getId();
+		inventory = medicalInventoryIoOperation.getInventoryById(inventoryId);
+		assertThat(inventory).isNotNull();
+		String status = InventoryStatus.validated.toString();
+		inventory.setStatus(status);
+		inventory = medicalInventoryIoOperation.updateMedicalInventory(inventory);
+		assertThat(inventory).isNotNull();
+		assertThat(inventory.getStatus()).isEqualTo(status);		
+	}
+	
+	@Test
+	void testConfirmMedicalInventory() throws Exception {
+		Ward ward = testWard.setup(false);
+		wardIoOperationRepository.saveAndFlush(ward);
+		MedicalInventory inventory = testMedicalInventory.setup(ward, false);
```

Comment:
you have to persist also chargeType and dischargeType as "inventory+" and "inventory-" are not found in the inmemory DB

Reply:
it seems that the supplier also is missing when it comes to create the charge movement

Reply:
I added also destination but I have the same error

Reply:
Fixed with 59237d65c8af98832088ea2c552157d5199cc5c2